autotest/utilities/test_gdal_translate.py

```diff
@@ -378,6 +378,27 @@ def test_gdal_translate_13(gdal_translate_path, tmp_path):
     ds = None
 
 
+###############################################################################
+# Test -dmo option
+
+
+def test_gdal_translate_dmo_option(gdal_translate_path, tmp_path):
+
+    dst_vrt = str(tmp_path / "test_dmo.vrt")
+
+    gdaltest.runexternal(
+        f"{gdal_translate_path} -dmo NEW_DOMAIN:META-TAG=value ../gcore/data/byte.tif {dst_vrt}"
```

Comment:
I'm going to have to do some homework for this python dictionary stuff, keys are unique right so it can' be 

```python
{ "domain_name" : { "key": "value" } , "domain_name" : { "key1": "value1" } }

```

but I guess a list of dictionaries is pythonic? 

```python
[ {"domain_name" : { "key": "value" }} , {"domain_name" : { "key1": "value1" }} ]
```

i'll explore what the api does now and the kinds of structures it deals with  🙏 this is good homework for me :)

Reply:
> I'm going to have to do some homework for this python dictionary stuff, keys are unique right so it can't be
> 
> ```python
> 
> ```

I was thinking to ``{ "domain1": { "key1": "value1", "key2": "value2" }, "domain2": .... } ``

Reply:
ah, lol ... right thanks!